app/book_list.rb

```diff
@@ -0,0 +1,85 @@
+require_relative './book'
+require_relative './label'
+require_relative '../data/book_crud'
+require_relative '../data/label_crud'
+
+module Booklist
+  def list_books
+    if @books.empty?
+      puts 'You don\'t have any Books.'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index} Title: #{book.name}, Publisher: #{book.publisher}, Publish_date:#{book.published_date},
+        Cover State: #{book.cover_state}"
+        puts ''
+      end
+    end
+    puts ''
+  end
+
+  def add_book
+    print 'Title: '
+    name = gets.chomp
+    print 'publisher: '
+    publisher = gets.chomp
+    print 'date of publish: '
+    published_date = gets.chomp
+    print 'cover state: '
+    cover_state = gets.chomp
+    stored_books = fetch_data('books')
+    label = handle_label
+    book = Book.new(publisher, cover_state, name, published_date)
+    @labels << label unless @labels.include?(label)
+    @books << book
+    book_data = { name: name, publisher: publisher, published_date: published_date,
+                  cover_state: cover_state }
+    stored_books.push(book_data)
+    update_data('books', stored_books)
+    puts 'Book added successfully'
+  end
+
+  def handle_label
+    if @labels.any?
+      print "enter 'N' to create a new label or 'S' to select an existing one"
+      option = gets.chomp.upcase
+      case option
+      when 'N'
+        add_label
+      when 'S'
+        puts 'select a label from the list by index'
+        list_labels
+        option = gets.chomp
+        @labels[option.to_i]
+      else
+        print 'invalid entry'
+      end
+    else
+      add_label
+    end
+  end
+
+  def list_labels
+    if @labels.empty?
+      puts 'You don\'t have any Labels.'
+    else
+      @labels.each_with_index do |label, index|
+        puts "#{index} Name: #{label.title}, Color: #{label.color}"
+        puts ''
+      end
+    end
+  end
+
+  def add_label
+    print 'Label Name: '
+    title = gets.chomp
+    print 'Color: '
+    color = gets.chomp
+    stored_label = fetch_data('labels')
+    label = Label.new(title, color)
+    @labels << label unless @labels.include?(label)
+    label_data = { name: title, color: color }
+    stored_label.push(label_data)
+    update_data('labels', stored_label)
+    puts 'Label added successfully'
+  end
```

Comment:
![image](https://user-images.githubusercontent.com/65068771/164893939-be0c219f-0c66-4a3c-a0b9-e727177f278d.png)

- [ ] Kindly fix this error :point_up:  after a book is created with a new label(not with an existing label), it is throwing an error when I am trying to list all genres. This is because you are saving the label with the `:title` but the saved it as `:name`. Hint :point_down: 
```rb 
# when you create a label with a title proprety
 label = Label.new(title, color)
```
```rb
# you save it in json with a name instead of a title
    label_data = { name: title, color: color }
```
```rb
# you read it with a title though you save it with a name
      @labels.each_with_index do |label, index|
        puts "#{index} Name: #{label.title}, Color: #{label.color}"
        puts ''
```